ppmclibs/tinycv_impl.cc

```diff
@@ -41,22 +41,6 @@ using namespace std;
 
 struct Image {
     cv::Mat img;
-    mutable cv::Mat _preped;
-
-    cv::Mat prep() const
-    {
-        if (!_preped.empty() && false)
```

Comment:
For the needles, prep'ing only the ROI still saves a considerable amount of work (typical needles are in the range of 10e3...100e3 pixels, while the whole screen is 800e3 pixels).

For the scene, it somewhat depends - for the regular checks, only the intersection area of the needles is required (plus margin), while for the 5 second full screen check it has to prep the whole scene. I think it is useful to do the fullscreen blur only if necessary.